Client/src/test/java/org/gluu/oxauth/ws/rs/deviceauthz/DeviceAuthzFlowHttpTest.java

```diff
@@ -0,0 +1,605 @@
+        assertNull(userInfoResponse.getClaim("work_phone"));
+    }
+
+    private TokenResponse processNewTokenWithRefreshToken(String scopes, String refreshToken, String clientId,
+                                                          String clientSecret) throws UnrecoverableKeyException,
+            NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+        TokenClient tokenClient2 = new TokenClient(tokenEndpoint);
+        tokenClient2.setExecutor(clientExecutor(true));
+        TokenResponse tokenResponse2 = tokenClient2.execRefreshToken(scopes, refreshToken, clientId, clientSecret);
+
+        showClient(tokenClient2);
+        assertEquals(tokenResponse2.getStatus(), 200, "Unexpected response code: " + tokenResponse2.getStatus());
+        assertNotNull(tokenResponse2.getEntity(), "The entity is null");
+        assertNotNull(tokenResponse2.getAccessToken(), "The access token is null");
+        assertNotNull(tokenResponse2.getTokenType(), "The token type is null");
+        assertNotNull(tokenResponse2.getRefreshToken(), "The refresh token is null");
+        assertNotNull(tokenResponse2.getScope(), "The scope is null");
+
+        return tokenResponse2;
+    }
+
+    private void verifyIdToken(String idToken) throws InvalidJwtException, UnrecoverableKeyException,
+            NoSuchAlgorithmException, KeyStoreException, KeyManagementException {
+        Jwt jwt = Jwt.parse(idToken);
+        assertNotNull(jwt.getHeader().getClaimAsString(JwtHeaderName.TYPE));
+        assertNotNull(jwt.getHeader().getClaimAsString(JwtHeaderName.ALGORITHM));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.ISSUER));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.AUDIENCE));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.EXPIRATION_TIME));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.ISSUED_AT));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.SUBJECT_IDENTIFIER));
+        assertNotNull(jwt.getClaims().getClaimAsString(JwtClaimName.OX_OPENID_CONNECT_VERSION));
+
+        RSAPublicKey publicKey = JwkClient.getRSAPublicKey(
+                jwksUri,
+                jwt.getHeader().getClaimAsString(JwtHeaderName.KEY_ID), clientExecutor(true));
+        RSASigner rsaSigner = new RSASigner(SignatureAlgorithm.RS256, publicKey);
+
+        assertTrue(rsaSigner.validate(jwt));
+    }
+
+    private TokenResponse processTokens(String clientId, String clientSecret, String deviceCode) {
+        TokenRequest tokenRequest = new TokenRequest(GrantType.DEVICE_CODE);
+        tokenRequest.setAuthUsername(clientId);
+        tokenRequest.setAuthPassword(clientSecret);;
+        tokenRequest.setAuthenticationMethod(AuthenticationMethod.CLIENT_SECRET_BASIC);
+        tokenRequest.setDeviceCode(deviceCode);
+
+        TokenClient tokenClient1 = newTokenClient(tokenRequest);
+        tokenClient1.setRequest(tokenRequest);
```

Comment:
This line is redundant, request already passed in `newTokenClient(tokenRequest);`